host/tests/unit/search-index-test.ts

```diff
@@ -743,6 +782,24 @@ posts/ignore-me.gts
       );
     });
 
+    test(`can search for cards that have custom queryableValue`, async function (assert) {
+      let matching = await indexer.search({
+        filter: {
+          on: { module: `${baseRealm.url}catalog-entry`, name: 'CatalogEntry' },
+          eq: {
+            ref: {
+              module: `${testModuleRealm}post`,
+              name: 'Post',
+            },
+          },
+        },
+      });
+      assert.deepEqual(
+        matching.map((m) => m.id),
+        [`${paths.url}catalog-entry-1`]
+      );
+    });
```

Comment:
Here's a test showing a query using an `ExportedCardRef` as the predicate of the query.